forage-android/src/main/java/com/joinforage/forage/android/ForageSDK.kt

```diff
@@ -98,10 +98,16 @@ class ForageSDK : ForageSDKInterface {
      *
      *         when (response) {
      *             is ForageApiResponse.Success -> {
-     *                 // parse response.data for the PaymentMethod object
+     *                 val paymentMethod = response.toPaymentMethod()
+     *                 // Unpack paymentMethod.ref, paymentMethod.card, etc.
+     *                 val card = paymentMethod.card
+     *                 // Unpack card.last4, ...
+     *                 val ebtCard = card as EbtCard
```

Comment:
*I'm not trying to propose any new designs here in this comment. I'm only trying to surface that the lack of reflection in this example snippet seems different from our discussion*

If we can cast `paymentMethod.card` as an `EbtCard` without doing any `if-checking`,  that means that `paymentMethod.card` is always an `EbtCard`. This behavior does not seem to align with our discussion yesterday. My understanding was that Forage would guarantee that `.card` would satisfy an `interface Card` so the merchant could access properties common to all cards for as long as possible. 

And if the merchant runs into a case where they absolutely must work with more specific information, their options would be:
1. do reflection to inspect `if (paymentMethod.card is EbtCard) { card as EbtCard }` or
2. Forage would expose something like `paymentMethod.card.ebt` to offer the more specific fields or
3. Forage would return something that is type specific (e.g. PaymentMethod<EbtCard>) 
4. Something else?